### ISSUE TITLE
Use read replica for raw queries

### ISSUE THREAD

**Issue by mellogarrett** [sequelize/sequelize#11649]
### Is your feature request related to a problem? Please describe.
When executing raw SQL queries, sequelize always uses the `write` host.

### Describe the solution you'd like
When you know that your raw SQL query is a read-only query, it would be great to be able to delegate to the read replica.

### Why should this be in Sequelize
Seems like a pretty common usecase that could benefit a lot of people. Limiting the read replica use to only queries generated from models is not ideal.

### Describe alternatives/workarounds you've considered
Can't think of any.

### Additional context
I tried to pass `{ type: QueryTypes.SELECT }` to the `.query` method thinking that might use the read replica but no luck.

### Is this issue dialect-specific?

- [x] No. This issue is relevant to Sequelize as a whole.
- [ ] Yes. This issue only applies to the following dialect(s): XXX, YYY, ZZZ

### Would you be willing to resolve this issue by submitting a Pull Request?

- [ ] Yes, I have the time and I know how to start.
- [x] Yes, I have the time but I don't know how to start, I would need guidance.
- [ ] No, I don't have the time, although I believe I could do it if I had the time...
- [ ] No, I don't have the time and I wouldn't even know how to start.

**Comment by papb** [comment]
Hello, thanks for the report!

> [X] Yes, I have the time but I don't know how to start, I would need guidance.

Nice, what kind of guidance do you need? I am not familiar with read replicas at all, I'm sure you know more than me about that, since you stumbled on this. How can I help you create a PR and get this solved? :grimacing:

**Comment by contra** [comment]
@mellogarrett FYI you can use `type: 'SELECT', useMaster: true` in the options as a workaround

**Comment by brianschardt** [comment]
@garrett-thompson i dont believe that work around actually works `userMaster: true` means it will not use the read replicas.
I would love this feature too.

**Comment by yocontra** [comment]
Please keep this open.

**Comment by berv63** [comment]
Can someone remove the existing workaround tag on this? Would also like to see this feature implemented and cannot find a suitable workaround.

**Comment by yocontra** [comment]
@berv63 The workaround I posted above (https://github.com/sequelize/sequelize/issues/11649#issuecomment-575443358) does work - it does require you to manually specify this on every read query.

**Comment by brightmileDaniel** [comment]
> @mellogarrett FYI you can use `type: 'SELECT', useMaster: false` in the options as a workaround

This doesn't seem to work on my side, any other ideas? Raw queries still spawn write connections unnecessarily before delegating to read instance. 

"sequelize": "6.29.0",

**Comment by ephys** [comment]
There is a bug in Sequelize 6. The "type" option that `sequelize.query` accepts is different from the "type" option that `getConnection` accepts. The former is "select, update, delete, etc" while the latter is "read, write"

This was fixed in Sequelize 7: https://github.com/sequelize/sequelize/blob/4d624fd37e84a0b9215bac0b0a1ab85892a81cef/packages/core/src/sequelize.js#L717

It was not fixed in Sequelize 6: https://github.com/sequelize/sequelize/blob/e4c780cb9478d4345a1742c26be0235d00501a03/src/sequelize.js#L638

**Comment by brightmileDaniel** [comment]
> There is a bug in Sequelize 6. The "type" option that `sequelize.query` accepts is different from the "type" option that `getConnection` accepts. The former is "select, update, delete, etc" while the latter is "read, write"
> 
> This was fixed in Sequelize 7:
> 
> https://github.com/sequelize/sequelize/blob/4d624fd37e84a0b9215bac0b0a1ab85892a81cef/packages/core/src/sequelize.js#L717
> 
> It was not fixed in Sequelize 6:
> 
> https://github.com/sequelize/sequelize/blob/e4c780cb9478d4345a1742c26be0235d00501a03/src/sequelize.js#L638

Thank you @ephys, is there a known workaround in version 6 or the only alternative is to update to version 7?

**Comment by ephys** [comment]
Actually I think I'm wrong: The connection manager also uses "SELECT" as the type in v6 (unlike v7, which uses read/write): https://github.com/sequelize/sequelize/blob/e4c780cb9478d4345a1742c26be0235d00501a03/src/dialects/abstract/connection-manager.js#L173

So after all, setting type to `'SELECT'` (uppercase) and either not setting `useMaster`, or setting it to `false`, should work

**Comment by naktinis** [comment]
@ephys just wanted to check the status of this. Is setting `type: 'SELECT'` currently enough to make sure a read replica is used and this issue could be closed? Or is this issue about a different feature?
